tensorflow/python/ops/image_ops_impl.py

```diff
@@ -1550,7 +1555,7 @@ def check_gif():
 
     def _png():
       """Decodes a PNG image."""
-      return gen_image_ops.decode_png(contents, channels)
+      return gen_image_ops.decode_png(contents, channels, dtype=dtype)
```

Comment:
Let's make this `dtype=tf.uint8 if dtype=tf.uint8 else tf.uint16`.

That combined with the unconditional call to `convert_image_tensor` will enable support for all dtypes for all image formats.